simulation/simple_sensor_simulator/test/src/test_scenario_simulator.cpp

```diff
@@ -0,0 +1,340 @@
+  auto request = simulation_api_schema::AttachLidarSensorRequest();
+  auto configuration = traffic_simulator::helper::constructLidarConfiguration(
+    traffic_simulator::helper::LidarType::VLP16, "entity_name", "awf/universe", 1.0);
+  *request.mutable_configuration() = configuration;
+  return request;
+}
+
+auto makeAttachOccupancyGridSensorRequest()
+  -> simulation_api_schema::AttachOccupancyGridSensorRequest
+{
+  auto request = simulation_api_schema::AttachOccupancyGridSensorRequest();
+  auto configuration = simulation_api_schema::OccupancyGridSensorConfiguration();
+
+  configuration.set_entity("entity_name");
+  configuration.set_architecture_type("awf/universe");
+  configuration.set_update_duration(1.0);
+  configuration.set_range(300.0);
+  configuration.set_resolution(1.0);
+  *request.mutable_configuration() = configuration;
+  return request;
+}
+
+class ScenarioSimulatorTest : public testing::Test
+{
+protected:
+  ScenarioSimulatorTest()
+  : server([] {
+      rclcpp::init(0, nullptr);
+      rclcpp::NodeOptions options;
+      auto component = std::make_shared<simple_sensor_simulator::ScenarioSimulator>(options);
+      rclcpp::spin(component);
+    }),
+    client(simulation_interface::TransportProtocol::TCP, "localhost", 5555U)
+  {
+  }
+  std::thread server;
+  zeromq::MultiClient client;
+};
+
+/**
+ * @note Test initialization correctness with a sample request with the default port (5555).
+ */
+TEST(ScenarioSimulator, initialize_defaultPort)
+{
+  auto server = std::thread([] {
+    rclcpp::init(0, nullptr);
+    rclcpp::NodeOptions options;
+    auto component = std::make_shared<simple_sensor_simulator::ScenarioSimulator>(options);
+    rclcpp::spin(component);
+  });
+
+  auto client =
+    zeromq::MultiClient(simulation_interface::TransportProtocol::TCP, "localhost", 5555U);
+
+  EXPECT_TRUE(client.call(makeInitializeRequest()).result().success());
+
+  rclcpp::shutdown();
+  server.join();
+}
+
+/**
+ * @note Test initialization correctness with a sample request with a custom port.
+ */
+TEST(ScenarioSimulator, initialize_customPort)
+{
+  auto server = std::thread([] {
+    rclcpp::init(0, nullptr);
+    rclcpp::NodeOptions options;
+    options.parameter_overrides().push_back(
+      rclcpp::Parameter{"port", rclcpp::ParameterValue(1234)});
+    auto component = std::make_shared<simple_sensor_simulator::ScenarioSimulator>(options);
+    rclcpp::spin(component);
+  });
+
+  auto client =
+    zeromq::MultiClient(simulation_interface::TransportProtocol::TCP, "localhost", 1234U);
+
+  EXPECT_TRUE(client.call(makeInitializeRequest()).result().success());
+
+  rclcpp::shutdown();
+  server.join();
+}
+
+/**
+ * @note Test updating frame correctness with a sample request.
+*/
+TEST_F(ScenarioSimulatorTest, updateFrame_correct)
+{
+  EXPECT_TRUE(client.call(makeInitializeRequest()).result().success());
+  EXPECT_TRUE(client.call(makeUpdateFrameRequest()).result().success());
+
+  rclcpp::shutdown();
+  server.join();
```

Comment:
consider moving these lines to destructor or TearDown(). And try to apply this to tests below